src/lib/oracleKeeperFetcher.ts

```diff
@@ -0,0 +1,286 @@
+      // eslint-disable-next-line no-console
+      console.error(`no available oracle keeper for chain ${this.chainId}`);
+      return;
+    }
+
+    // eslint-disable-next-line no-console
+    console.log(`switch oracle keeper to ${getOracleKeeperUrl(this.chainId, nextIndex)}`);
+
+    this.setOracleKeeperInstancesConfig((old) => {
+      return { ...old, [this.chainId]: nextIndex };
+    });
+
+    fallbackThrottleTimerId = setTimeout(() => {
+      fallbackThrottleTimerId = undefined;
+    }, 5000);
+  }
+
+  fetchTickers(): Promise<TickersResponse> {
+    return fetch(buildUrl(this.url!, "/prices/tickers"))
+      .then((res) => res.json())
+      .then((res) => {
+        if (!res.length) {
+          throw new Error("Invalid tickers response");
+        }
+
+        return res;
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.error(e);
+        this.switchOracleKeeper();
+
+        throw e;
+      });
+  }
+
+  fetch24hPrices(): Promise<DayPriceCandle[]> {
+    return fetch(buildUrl(this.url!, "/prices/24h"))
+      .then((res) => res.json())
+      .then((res) => {
+        if (!res?.length) {
+          throw new Error("Invalid 24h prices response");
+        }
+
+        return res;
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.error(e);
+        this.switchOracleKeeper();
+        throw e;
+      });
+  }
+
+  fetchPostReport(body: { report: object; version: string | undefined; isError: boolean }): Promise<Response> {
+    return fetch(buildUrl(this.url!, "/report/ui"), {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
+    });
+  }
+
+  fetchPostReport2(body: PostReport2Body, debug): Promise<Response> {
+    if (debug) {
+      // eslint-disable-next-line no-console
+      console.log("sendMetric", body.event, body);
+    }
+
+    return fetch(buildUrl(this.url!, "/report/ui2"), {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
+    });
+  }
+
+  fetchPostTimingReport(body: { event: string; time: number }): Promise<Response> {
+    return fetch(buildUrl(this.url!, "/report/ui/timing"), {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        ...body,
+        isDev: isDevelopment(),
+        host: window.location.host,
+        version: APP_VERSION,
+      }),
+    });
+  }
+
+  fetchPostEventReport(body: { event: string }): Promise<Response> {
+    return fetch(buildUrl(this.url!, "/report/ui/counter"), {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        ...body,
+        isDev: isDevelopment(),
+        host: window.location.host,
+        version: APP_VERSION,
```

Comment:
i have just realized we also need to send `abFlags` to this endpoint, let's add it to the request body

otherwise we won't be able to compare data from different groups in datadog

Reply:
the endpoint accepts abFlags as Record<string, boolean>